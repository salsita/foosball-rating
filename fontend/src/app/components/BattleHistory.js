import React, { Component } from 'react';

// components pure html
import {
  Title, Subtitle, TextSpan,
  ListCon, ListItem,
  ALink,
  GridContainer
} from './../../styles/blocks/';

class BattleHistory extends Component {
  render() {
    return(

      <ListCon>

        {/* One Item of list */}
        <ListItem Display="grid" Column="2fr 1fr 2fr">
          <TextSpan textAlign="right">
            <ALink>Jhon</ALink>
            & 
            <ALink>Jack</ALink>
          </TextSpan>
          <TextSpan textAlign="center">VS</TextSpan>
          <TextSpan textAlign="left">
            <ALink>Tom</ALink>
            & 
            <ALink>Nick</ALink>
          </TextSpan>
        </ListItem>

        <ListItem Display="grid" Column="2fr 1fr 2fr">
          <TextSpan textAlign="right">
            <ALink>Jhon</ALink>
            & 
            <ALink>Jack</ALink>
          </TextSpan>
          <TextSpan textAlign="center">VS</TextSpan>
          <TextSpan textAlign="left">
            <ALink>Tom</ALink>
            & 
            <ALink>Nick</ALink>
          </TextSpan>
        </ListItem>


      </ListCon>
      
    )
  }
}

export default BattleHistory;