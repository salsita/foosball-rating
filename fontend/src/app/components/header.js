import React, { Component } from 'react';

// components pure html
import {
    Title, Subtitle
} from './../../styles/blocks/';

class Header extends Component {
    render() {
        return(
            <div>
                <Title>
                    Logo
                </Title>
                <Subtitle>
                    Powered by Salsita
                </Subtitle>
            </div>
        )
    }
}

export default Header;