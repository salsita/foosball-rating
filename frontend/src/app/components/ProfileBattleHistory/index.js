import React from 'react'
import { ListCon } from './../../../styles/blocks'
import { ProfileBattleHistoryRow } from './ProfileBattleHistoryRow'
import { TextSpan } from '../../../styles/blocks/typo'

const ProfileBattleHistory = props => (
  <>
    <TextSpan>
      Match history
    </TextSpan>
    <ListCon>
      {props.matches.map(match =>
        <ProfileBattleHistoryRow key={match.id} match={match} />,
      )}
    </ListCon>
  </>
)

export default ProfileBattleHistory
