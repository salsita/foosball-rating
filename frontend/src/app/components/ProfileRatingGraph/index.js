import React from 'react'
import { connect } from 'react-redux'
import {
  FlexibleWidthXYPlot,
  XAxis,
  YAxis,
  HorizontalGridLines,
  VerticalGridLines,
  LineSeries,
} from 'react-vis'

import { getRatingHistoryGraphForPlayer } from '../../modules/matches/matches-selectors'
import { Box, TextSpan } from '../../../styles/blocks'
import { plotAxisStyle, plotGridStyle, plotLineStyle, plotMainGridStyle } from '../../../styles/svg'

const SimpleProfileRatingGraph = props => (
  <Box Margin='0 5px 20px 5px'>
    <TextSpan>
      Rating history
    </TextSpan>
    <div>
      <FlexibleWidthXYPlot
        height={300}
        xType='time'
        margin={{ left: 45 }}
      >
        <HorizontalGridLines
          style={plotGridStyle}
        />
        <HorizontalGridLines
          style={plotMainGridStyle}
          tickValues={[1000]}
        />
        <VerticalGridLines />
        <XAxis
          style={plotAxisStyle}
          tickLabelAngle={-45}
        />
        <YAxis style={plotAxisStyle} />
        <LineSeries
          data={props.ratingHistoryGraph}
          style={plotLineStyle}
        />
      </FlexibleWidthXYPlot>
    </div>
  </Box>
)


const mapStateToProps = (state, props) => ({
  ratingHistoryGraph: getRatingHistoryGraphForPlayer(state, props.playerId),
})

export const ProfileRatingGraph = connect(mapStateToProps)(SimpleProfileRatingGraph)
