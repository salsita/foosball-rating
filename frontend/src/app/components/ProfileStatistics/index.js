import React from 'react'

import { TextSpan, StatisticsTable, StatisticsValue, StyledLink, WinnerSpan } from '../../../styles/blocks'
import { createProfilePath } from '../../const/routes'

const ProfileStatistics = props => {
  const {
    rankings: { ranking, toNextRank, toPrevRank },
    statistics: { 
      bestDay, worstDay, mostFrequentTeammate, leastFrequentTeammate, mostFrequentOpponent, 
      leastFrequentOpponent, mostSuccessTeammate, leastSuccessTeammate, mostSuccessOpponent, 
      leastSuccessOpponent,
    },
  } = props

  return (
    <>
      <TextSpan>Ranking: {ranking}</TextSpan>
      <StatisticsTable>
        <tbody>
          <tr>
            <td>To Next Rank: 
              <StatisticsValue>{getScorePart(toNextRank, toNextRank <= 20, '+')}</StatisticsValue>
            </td>
            <td>To Prev Rank: 
              <StatisticsValue>{getScorePart(toPrevRank, toPrevRank > 20, '-')}</StatisticsValue>
            </td>
          </tr>
          <tr>
            <td>Best Day: 
              <StatisticsValue>{bestDay?.date} {getDayScorePart(bestDay?.value)}</StatisticsValue>
            </td>
            <td>Worst Day: 
              <StatisticsValue>{worstDay?.date} {getDayScorePart(worstDay?.value)}</StatisticsValue>
            </td>
          </tr>
          <tr>
            {getUserRow('Most Played With', mostFrequentTeammate, 'matches', true)}
            {getUserRow('Least PLayed With', leastFrequentTeammate, 'matches', false)}
          </tr>
          <tr>
            {getUserRow('Most Played Against', mostFrequentOpponent, 'matches', true)}
            {getUserRow('Least Played Against', leastFrequentOpponent, 'matches', false)}
          </tr>
          <tr>
            {getUserRow('Most Won With', mostSuccessTeammate, 'wins', true)}
            {getUserRow('Least Won With', leastSuccessTeammate, 'wins', false)}
          </tr>
          <tr>
            {getUserRow('Most Won Over', mostSuccessOpponent, 'wins', true)}
            {getUserRow('Most Lost To', leastSuccessOpponent, 'losses', false)}
          </tr>
        </tbody>
      </StatisticsTable>
    </>
  )
}

const getDayScorePart = score => getScorePart(score, score > 0, score > 0 ? '+' : '')

const getScorePart = (value, isGreen, sign = 'x') => (
  <WinnerSpan winner={isGreen}>{sign}{value}</WinnerSpan>
)

const getUserRow = (text, stat, valueKey, isGreen) => (
<td>{text}: 
  <StatisticsValue>
    {getPlayerLink(stat.user)} {getScorePart(stat[valueKey], isGreen)}
  </StatisticsValue>
</td>
)

const getPlayerLink = player => (
  <StyledLink key={player?.id} to={createProfilePath(player?.id)}>{player?.name} 
   &nbsp;<span>({player?.matchRating})</span></StyledLink>
)

export default ProfileStatistics
