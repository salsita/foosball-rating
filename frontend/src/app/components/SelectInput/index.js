import React from 'react'
import { SelectBox } from './../../../styles/blocks'

const SelectInput = ({ items, selectedValue, onChange }) => (
  <SelectBox onChange={onChange} value={selectedValue}>
    {items.map(item =>
      <option key={item.value} value={item.value}>
        {item.label}
      </option>,
    )}
  </SelectBox>
)

export default SelectInput


