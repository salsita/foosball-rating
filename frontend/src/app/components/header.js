import React, { Component } from 'react';
import {
  Title, Subtitle,
  Logo,
  Nav
} from './../../styles/blocks/';

const logo = require('./../../media/logo.png');

class Header extends Component {
  render() {
    return(
      <Nav>
        <Logo href="https://www.salsitasoft.com/"><img src={logo} /></Logo>
        <Title>ELO rating</Title>
      </Nav>
    )
  }
}

export default Header;
