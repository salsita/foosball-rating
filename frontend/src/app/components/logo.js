import React from 'react'
import logoSmall from './../../media/logo-small.png'
import logo from './../../media/logo.png'
import { Link } from 'react-router-dom'

export const LogoComponent = ({ className }) => (
  <span className={className}>
    <Link id='smalllogo' to={'/'}>
      <img src={logoSmall} alt="logo" />
    </Link>
    <Link id='largelogo' to={'/'}>
      <img src={logo} alt="logo" />
    </Link>
  </span>
)
