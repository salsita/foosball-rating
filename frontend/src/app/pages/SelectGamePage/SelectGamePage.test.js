import { SelectGamePageComponent } from './index'
import { SelectInput } from '../../components/SelectInput'
import { HISTORY_MOCK } from '../../../tests/mocks'
import { shallowWithProps } from '../../../tests/utils'
import { FOOSBALL_GAME, URLS } from '../../../tests/data'

describe('SelectGamePageComponent', () => {
  let selectGamePage
  beforeEach(() => {
    jest.clearAllMocks()
  })
  describe('when there is a foosball game', () => {
    beforeEach(() => {
      selectGamePage = shallowWithProps(SelectGamePageComponent, {
        games: [ FOOSBALL_GAME ],
        history: HISTORY_MOCK,
      })
    })
    it('renders Header and select', () => {
      expect(selectGamePage).toMatchSnapshot()
    })
    describe('when foosball is selected', () => {
      beforeEach(() => {
        selectGamePage.find(SelectInput).simulate('change', {
          target: { value: FOOSBALL_GAME.name },
        })
      })
      it('push a redirect to history', () => {
        expect(HISTORY_MOCK.push).toBeCalledWith(URLS.FOOSBALL)
      })
    })
  })
})
