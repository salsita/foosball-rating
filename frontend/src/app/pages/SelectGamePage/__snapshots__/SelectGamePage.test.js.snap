// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGamePageComponent when there is a foosball game renders Header and select 1`] = `
<Fragment>
  <Connect(HeaderComponent) />
  <styled.div>
    <styled.h2>
      Games
    </styled.h2>
    <SelectInput
      items={
        Array [
          Object {
            "label": "Select the game!",
            "value": "Select the game!",
          },
          Object {
            "label": "foosball",
            "value": "foosball",
          },
        ]
      }
      onChange={[Function]}
    />
  </styled.div>
  <Footer />
</Fragment>
`;
